save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/OrganizationController.kt

```diff
@@ -362,8 +367,17 @@ internal class OrganizationController(
         .switchIfEmptyToResponseException(HttpStatus.FORBIDDEN) {
             "Not enough permission for managing organization git credentials."
         }
-        .map {
-            gitService.delete(it, url)
+        .map { organization ->
+            // Find and remove all corresponding data to the current git repository
+            val git = gitService.getByOrganizationAndUrl(organization, url)
+            val testSuitesSources = testSuitesSourceService.findByGit(git)
+            testSuitesSources.forEach { testSuitesSource ->
+                val testSuites = testSuitesService.getBySource(testSuitesSource)
+                testSuitesService.deleteTestSuiteDto(testSuites.map { it.toDto() })
```

Comment:
Also, maybe check that there is no Execution in 'PENDING' state with TestSuites selected to be removed

Reply:
Actually, we didn't remove executions at all, fixed

Reply:
I'm about case when Execution is created with some test suites source but not started yet and then someone else remove TestSuitesSource.
I think it's rare case